components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -647,6 +649,69 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
            !$omp end parallel
          end if
 
+         ! When config_GM_closure = 'Griffies' actually modify the value of gmBolusKappa based on
+         ! Griffies et al (2005)
+         if (local_config_GM_compute_Griffies) then
+
+            ! This is Griffies et al. (2005) eqn 15:
+            ! A_{gm} = alpha |grad_z rho| L^2 g / ( rho_0 N_0 )
+            ! here
+            ! coef = alpha L^2 g / ( rho_0 N_0 )
+            ! A_{gm} = coef * |grad_z rho|
+            coef = config_GM_Griffies_alpha * config_gm_Griffies_length**2 * gravity &
+                   / (config_GM_Griffies_ref_density*config_GM_Griffies_No + 1.0E-15)
+
+            !$omp parallel
+            !$omp do schedule(runtime) private(k, cell1, cell2, sum_gradDensity, ltSum, countN2, &
+            !$omp             zEdge, ltAvg)
+            do iEdge = 1, nEdges
+               k=minLevelEdgeBot(iEdge)
+               sum_gradDensity = 0.0_RKIND
+               ltSum = 0.0_RKIND
+               countN2 = 0
+               cell1 = cellsOnEdge(1, iEdge)
+               cell2 = cellsOnEdge(2, iEdge)
+
+               k = minLevelEdgeBot(iEdge)+1
+               zEdge = -layerThickEdgeMean(k-1,iEdge)
+               do while(zEdge > -config_GM_Griffies_min_depth .and. k < maxLevelEdgeTop(iEdge))
+                  k = k+1
+                  zEdge = zEdge - layerThickEdgeMean(k-1,iEdge)
+               end do
+
+               do while(zEdge > -config_GM_Griffies_max_depth .and. k < maxLevelEdgeTop(iEdge))
+                  ltAvg = 0.25_RKIND*(   layerThickness(k,cell1) + layerThickness(k-1,cell1) &
+                                       + layerThickness(k,cell2) + layerThickness(k-1,cell2))
+                  sum_gradDensity = sum_gradDensity + abs(gradDensityConstZTopOfEdge(k,iEdge))*ltAvg
+                  ltsum = ltsum + ltAvg
+                  countN2 = countN2 + 1
+
+                  k = k + 1
+                  zEdge = zEdge - layerThickEdgeMean(k-1,iEdge)
+               end do
+
+               if (countN2 > 0) then
+                  sum_gradDensity = sum_gradDensity / ltsum
+
+                  ! This is Griffies et al. (2005) eqn 15:
+                  ! A_{gm} = alpha |grad_z rho| L^2 g / ( rho_0 N_0)
+                  gmBolusKappa(iEdge) = coef * sum_gradDensity
+
+                  ! These were the original lines from Lukes cryo branch:
+                  ! gmBolusKappa(iEdge) = config_GM_Griffies_alpha * config_gm_Griffies_length**2.0_RKIND * &
+                  !                      gravity * abs(sumN2) / (1.0E-15 + config_GM_Griffies_No)
+                  ! note that rho_0 is missing from the denominator.
```

Comment:
@vanroekel please look carefully at this. Your original lines are above, also here: https://github.com/E3SM-Project/E3SM/blob/9235712a8b87197a2b745fc55fb5ed24bf54e946/components/mpas-ocean/src/shared/mpas_ocn_gm.F#L678-L679

Compare to Griffies et al. (2005) eqn 15, you are missing the `rho_0` in the denominator. So I added it with another flag, since he uses 1035 while we use `config_density0=1026`. If your original line was missing `1/rho_0` then `gmBolusKappa` would have been too small by 1000, so results would be similar to turning off GM.

Reply:
@mark-petersen yes that rho_o absolutely should be there.  I know I checked the values at one point so that change must not have gotten pushed to you.  The units don't work out to m2/s without it, so support adding rho_o to the denominator

Reply:
@vanroekel thanks for reviewing. Agreed, let's just use `config_density0` here. I just revised.